lib/SILOptimizer/Mandatory/PredictableMemOpt.cpp

```diff
@@ -3169,9 +3169,12 @@ static AllocationInst *getOptimizableAllocation(SILInstruction *i) {
 }
 
 bool swift::optimizeMemoryAccesses(SILFunction *fn) {
+  if (!fn->hasOwnership()) {
```

Comment:
Did fixing the presence of the flag reveal some new issues?

Reply:
Yeah. Now that Synchronization and Distributed are serialized in non-ossa, I had to add these bailouts since we run mandatory optimizations on serialized functions as well.